api/src/test/java/org/motechproject/nms/api/osgi/KilkariControllerBundleIT.java

```diff
@@ -212,22 +220,145 @@ public void testCreateSubscriptionRequest() throws IOException, InterruptedExcep
+        mctsSubscriber.setDateOfBirth(LocalDate.now().minusDays(14));
+        subscriberDataService.create(mctsSubscriber);
+
+        subscriptionService.createSubscription(9999911122L, "10", "pack1", SubscriptionMode.MCTS_IMPORT);
+
+        mctsSubscriber = subscriberDataService.findByCallingNumber(9999911122L);
+        assertEquals(1, mctsSubscriber.getSubscriptions().size());
+    }
+
+    @Test
+    public void testCreateDuplicateChildSubscriptionViaMcts() {
+        setupData();
+
+        Subscriber mctsSubscriber = new Subscriber(9999911122L);
+        mctsSubscriber.setDateOfBirth(LocalDate.now().minusDays(14));
+        subscriberDataService.create(mctsSubscriber);
+
+        subscriptionService.createSubscription(9999911122L, "10", "pack1", SubscriptionMode.MCTS_IMPORT);
+
+        // attempt to create subscription to the same pack -- should fail
+        subscriptionService.createSubscription(9999911122L, "10", "pack1", SubscriptionMode.MCTS_IMPORT);
+
+        mctsSubscriber = subscriberDataService.findByCallingNumber(9999911122L);
+        assertEquals(1, mctsSubscriber.getSubscriptions().size());
+    }
+
+    @Test
+    public void testCreateDuplicatePregnancySubscriptionViaMcts() {
+        setupData();
+
+        Subscriber mctsSubscriber = new Subscriber(9999911122L);
+        mctsSubscriber.setLastMenstrualPeriod(LocalDate.now().minusDays(28));
+        subscriberDataService.create(mctsSubscriber);
+
+        subscriptionService.createSubscription(9999911122L, "10", "pack2", SubscriptionMode.MCTS_IMPORT);
+
+        // attempt to create subscription to the same pack -- should fail
+        subscriptionService.createSubscription(9999911122L, "10", "pack2", SubscriptionMode.MCTS_IMPORT);
+
+        mctsSubscriber = subscriberDataService.findByCallingNumber(9999911122L);
+        assertEquals(1, mctsSubscriber.getSubscriptions().size());
+    }
+
+    @Test
+    public void testCreateSecondPregnancySubscriptionAfterDeactivationViaMcts() {
+        setupData();
+
+        Subscriber mctsSubscriber = new Subscriber(9999911122L);
+        mctsSubscriber.setLastMenstrualPeriod(LocalDate.now().minusDays(28));
+        subscriberDataService.create(mctsSubscriber);
+
+        subscriptionService.createSubscription(9999911122L, "10", "pack2", SubscriptionMode.MCTS_IMPORT);
+        mctsSubscriber = subscriberDataService.findByCallingNumber(9999911122L);
+        Subscription pregnancySubscription = mctsSubscriber.getSubscriptions().iterator().next();
+        pregnancySubscription.setStatus(SubscriptionStatus.DEACTIVATED);
+        subscriptionDataService.update(pregnancySubscription);
+
+        // attempt to create subscription to the same pack -- should succeed
+        subscriptionService.createSubscription(9999911122L, "10", "pack2", SubscriptionMode.MCTS_IMPORT);
+
+        mctsSubscriber = subscriberDataService.findByCallingNumber(9999911122L);
+        assertEquals(2, mctsSubscriber.getSubscriptions().size());
```

Comment:
I'm a little concerned that we might return deactivated subscriptions back to the IVR.  Should we split getSubscriptions into getActiveSubscriptions and getAllSubscriptions?